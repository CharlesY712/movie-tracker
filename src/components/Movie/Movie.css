.movie {
  position: relative;
  vertical-align: top;
  width: 400px;
  height: 200px;
  background-color: var(--card-bg);
  border: 3px solid gray;
  border-radius: 30px;
  margin: 5px;
  overflow-y: hidden;
  padding: 10px 20px 20px 20px;
}
.movie p {
  width: 80%;
  margin: 4px auto;
}
#title {
  display: block;
  margin: 0;
}
.movie-styling {
  display: flex;
  height: 100%;
}

#image {
  display: inline-flex;
  height: 200px;
  /* width: 185px; */
}
.movie-info-container {
  overflow-y: auto;
  height: 100%;
}
.favorites {
  background-color: var(--card-bg);
  position: absolute;
  top: 10px;
  right: 25px;
  width: 50px;
  height: 50px;
  font-size: 3em;
  border-radius: 50%;
  border: none;
  color: var(--fave-color);
}

#favs {
  color: green;
}

::-webkit-scrollbar {
    width: 10px;
}

::-webkit-scrollbar-track {
    display: none;
}

