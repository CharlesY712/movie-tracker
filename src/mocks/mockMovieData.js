module.exports = {results: [{
adult:false,
id:337167,
overview: "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
popularity:572.203541,
poster_path:"/jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
release_date:"2018-02-07",
title:"Fifty Shades Freed",
video:false,
vote_average:6.1,
vote_count:1097
},
{
adult:false,
id:284054,
overview:"King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan special forces), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
popularity:329.754985,
poster_path:"/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
release_date:"2018-02-13",
title:"Black Panther",
video:false,
vote_average:7.4,
vote_count:3477
}]}
