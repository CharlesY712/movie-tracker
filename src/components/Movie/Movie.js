import React from 'react';
import PropTypes from 'prop-types';
import { postFavorites, getFavorites } from '../../helper/apiCall';
import './Movie.css';

const Movie = ({movie, user, addFavorites}) => {
  const {
    title,
    // eslint-disable-next-line
    id,
    rating,
    image,
    date,
    summary} = movie;

  const handleClick = () => {
    !user.length ?
      alert('Please LogIn or Create An Account!')
      :
      createFavorites();
  };

  const createFavorites = async () => {
    const userFavorites = await getFavorites(user[0].id)
    if (!userFavorites.error) {
      const  favorites = userFavorites.find(fav => fav.movie_id === movie.id)
      console.log({favorites})
      console.log({id})
      favorites ? alert('You already love this movie!') : await postFavorites(movie, user[0].id) 
    }
    return
  };

  return (
    <article className='movie'>
      <button className='favorites'onClick={handleClick}>❤︎</button>
      <h1 className='title'>{title}</h1>
      <img id='image' src={image} alt='movie poster'/>
      <p className='rating'>{rating}</p>
      <p className='date'>{date}</p>
      <p className='summary'>Summary: {summary}</p>
    </article>
  );
};

Movie.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.array
};

export default Movie;
