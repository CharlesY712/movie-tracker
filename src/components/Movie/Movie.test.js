import React from 'react';
import { mount, shallow } from 'enzyme';
import Movie from './Movie';
import * as mocks from '../../mocks/mockMovieData'

describe('Movie', () => {
  let wrapper;
  let mockUser;
  let mockHandleUser;
  let favsMovie;
  let mockKey;

  beforeEach(() => {
    mockUser = [{
      id: 1,
      name: "Bob",
      password: "ilovemywife",
      email: "billybob@gmail.com",
      favorites: [mocks.cleanData]
    }]
    mockHandleUser = jest.fn();
    favsMovie = true;
    mockKey = 337167;

    wrapper = shallow(
      <Movie
        key={mockKey}
        movie={mocks.cleanMovie}
        user={mockUser}
        handleUser={mockHandleUser}
        favsMovie={favsMovie}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke handleClick to create an alert if there is not a user', () => {
    const user = [];
    wrapper = shallow(
      <Movie
        key={mockKey}
        movie={mocks.cleanMovie}
        user={user}
        handleUser={mockHandleUser}
        favsMovie={favsMovie}
      />)

    wrapper.find('button').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  it('should call getFavorites', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve(mocks.cleanData));
    wrapper.instance().createFavorites();
    await wrapper.update();
    const expected = await wrapper.updateFavorites(mockUser.favorites)

    expect(expected).toHaveBeenCalled()
  });


});
