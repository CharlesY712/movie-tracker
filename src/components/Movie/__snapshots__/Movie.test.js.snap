// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should call getFavorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    favsMovie={true}
    handleUser={[Function]}
    movie={
        Object {
            "movie_id": 337167,
            "overview": "Believing they have left behind...",
            "poster_path": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
            "release_date": "2018-02-07",
            "title": "Fifty Shades Freed",
            "vote_average": 6.1,
          }
    }
    user={
        Array [
            Object {
              "person": "bob",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorites"
          id={true}
          onClick={[Function]}
>
          ❤︎
</button>,
        <h1
          className="title"
>
          Fifty Shades Freed
</h1>,
        <img
          alt="movie poster"
          id="image"
          src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
        <p
          className="rating"
>
          6.1
</p>,
        <p
          className="date"
>
          2018-02-07
</p>,
        <p
          className="summary"
>
          Summary: 
          Believing they have left behind...
</p>,
      ],
      "className": "movie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "❤︎",
          "className": "favorites",
          "id": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "❤︎",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fifty Shades Freed",
          "className": "title",
        },
        "ref": null,
        "rendered": "Fifty Shades Freed",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "id": "image",
          "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 6.1,
          "className": "rating",
        },
        "ref": null,
        "rendered": 6.1,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-02-07",
          "className": "date",
        },
        "ref": null,
        "rendered": "2018-02-07",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "className": "summary",
        },
        "ref": null,
        "rendered": Array [
          "Summary: ",
          "Believing they have left behind...",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorites"
            id={true}
            onClick={[Function]}
>
            ❤︎
</button>,
          <h1
            className="title"
>
            Fifty Shades Freed
</h1>,
          <img
            alt="movie poster"
            id="image"
            src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
          <p
            className="rating"
>
            6.1
</p>,
          <p
            className="date"
>
            2018-02-07
</p>,
          <p
            className="summary"
>
            Summary: 
            Believing they have left behind...
</p>,
        ],
        "className": "movie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "❤︎",
            "className": "favorites",
            "id": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "❤︎",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fifty Shades Freed",
            "className": "title",
          },
          "ref": null,
          "rendered": "Fifty Shades Freed",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "id": "image",
            "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 6.1,
            "className": "rating",
          },
          "ref": null,
          "rendered": 6.1,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-02-07",
            "className": "date",
          },
          "ref": null,
          "rendered": "2018-02-07",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Summary: ",
              "Believing they have left behind...",
            ],
            "className": "summary",
          },
          "ref": null,
          "rendered": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Movie should invoke handleClick to create an alert if there is not a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    favsMovie={true}
    handleUser={[Function]}
    movie={
        Object {
            "movie_id": 337167,
            "overview": "Believing they have left behind...",
            "poster_path": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
            "release_date": "2018-02-07",
            "title": "Fifty Shades Freed",
            "vote_average": 6.1,
          }
    }
    user={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorites"
          id={true}
          onClick={[Function]}
>
          ❤︎
</button>,
        <h1
          className="title"
>
          Fifty Shades Freed
</h1>,
        <img
          alt="movie poster"
          id="image"
          src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
        <p
          className="rating"
>
          6.1
</p>,
        <p
          className="date"
>
          2018-02-07
</p>,
        <p
          className="summary"
>
          Summary: 
          Believing they have left behind...
</p>,
      ],
      "className": "movie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "❤︎",
          "className": "favorites",
          "id": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "❤︎",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fifty Shades Freed",
          "className": "title",
        },
        "ref": null,
        "rendered": "Fifty Shades Freed",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "id": "image",
          "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 6.1,
          "className": "rating",
        },
        "ref": null,
        "rendered": 6.1,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-02-07",
          "className": "date",
        },
        "ref": null,
        "rendered": "2018-02-07",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "className": "summary",
        },
        "ref": null,
        "rendered": Array [
          "Summary: ",
          "Believing they have left behind...",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorites"
            id={true}
            onClick={[Function]}
>
            ❤︎
</button>,
          <h1
            className="title"
>
            Fifty Shades Freed
</h1>,
          <img
            alt="movie poster"
            id="image"
            src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
          <p
            className="rating"
>
            6.1
</p>,
          <p
            className="date"
>
            2018-02-07
</p>,
          <p
            className="summary"
>
            Summary: 
            Believing they have left behind...
</p>,
        ],
        "className": "movie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "❤︎",
            "className": "favorites",
            "id": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "❤︎",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fifty Shades Freed",
            "className": "title",
          },
          "ref": null,
          "rendered": "Fifty Shades Freed",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "id": "image",
            "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 6.1,
            "className": "rating",
          },
          "ref": null,
          "rendered": 6.1,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-02-07",
            "className": "date",
          },
          "ref": null,
          "rendered": "2018-02-07",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Summary: ",
              "Believing they have left behind...",
            ],
            "className": "summary",
          },
          "ref": null,
          "rendered": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Movie should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    favsMovie={true}
    handleUser={[Function]}
    movie={
        Object {
            "movie_id": 337167,
            "overview": "Believing they have left behind...",
            "poster_path": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
            "release_date": "2018-02-07",
            "title": "Fifty Shades Freed",
            "vote_average": 6.1,
          }
    }
    user={
        Array [
            Object {
              "email": "billybob@gmail.com",
              "favorites": Array [
                Array [
                  Object {
                    "movie_id": 337167,
                    "overview": "Believing they have left behind...",
                    "poster_path": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
                    "release_date": "2018-02-07",
                    "title": "Fifty Shades Freed",
                    "vote_average": 6.1,
                  },
                  Object {
                    "movie_id": 284054,
                    "overview": "King T'Challa returns home....",
                    "poster_path": "http://image.tmdb.org/t/p/w185//uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                    "release_date": "2018-02-13",
                    "title": "Black Panther",
                    "vote_average": 7.4,
                  },
                ],
              ],
              "id": 1,
              "name": "Bob",
              "password": "ilovemywife",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorites"
          id={true}
          onClick={[Function]}
>
          ❤︎
</button>,
        <h1
          className="title"
>
          Fifty Shades Freed
</h1>,
        <img
          alt="movie poster"
          id="image"
          src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
        <p
          className="rating"
>
          6.1
</p>,
        <p
          className="date"
>
          2018-02-07
</p>,
        <p
          className="summary"
>
          Summary: 
          Believing they have left behind...
</p>,
      ],
      "className": "movie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "❤︎",
          "className": "favorites",
          "id": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "❤︎",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fifty Shades Freed",
          "className": "title",
        },
        "ref": null,
        "rendered": "Fifty Shades Freed",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "id": "image",
          "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 6.1,
          "className": "rating",
        },
        "ref": null,
        "rendered": 6.1,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-02-07",
          "className": "date",
        },
        "ref": null,
        "rendered": "2018-02-07",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "className": "summary",
        },
        "ref": null,
        "rendered": Array [
          "Summary: ",
          "Believing they have left behind...",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorites"
            id={true}
            onClick={[Function]}
>
            ❤︎
</button>,
          <h1
            className="title"
>
            Fifty Shades Freed
</h1>,
          <img
            alt="movie poster"
            id="image"
            src="http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg"
/>,
          <p
            className="rating"
>
            6.1
</p>,
          <p
            className="date"
>
            2018-02-07
</p>,
          <p
            className="summary"
>
            Summary: 
            Believing they have left behind...
</p>,
        ],
        "className": "movie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "❤︎",
            "className": "favorites",
            "id": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "❤︎",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fifty Shades Freed",
            "className": "title",
          },
          "ref": null,
          "rendered": "Fifty Shades Freed",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "id": "image",
            "src": "http://image.tmdb.org/t/p/w185//jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 6.1,
            "className": "rating",
          },
          "ref": null,
          "rendered": 6.1,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-02-07",
            "className": "date",
          },
          "ref": null,
          "rendered": "2018-02-07",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Summary: ",
              "Believing they have left behind...",
            ],
            "className": "summary",
          },
          "ref": null,
          "rendered": Array [
            "Summary: ",
            "Believing they have left behind...",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
